### ISSUE TITLE
Focus outline missing on top-level menu bar items upon reaching them by Esc key press

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5#16719]
## 📝 Provide detailed reproduction steps (if any)

1. Open menu bar.
2. Go somewhere deep.
3. Start escaping using Esc.

### ✔️ Expected result

When you reach the top-level category such as "Insert", the button should have a focus outline.

### ❌ Actual result

The focus outline is not there. You have to start navigating horizontally (arrow left/right) to get focus outlines on top-level categories.

https://github.com/user-attachments/assets/cd16d4ae-d33a-4d86-8e1f-00a102a15185

## 📃 Other details

I suppose this is an aftermath of hacks that were made to keep the focus outline invisible until the user starts keyboard navigation in the menu bar.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.
